File: Valid Paranthesis.cpp
#include<stack>
#include<cstring>
class Solution {
public:
    bool isValid(string str) {
        if(str.length()%2!=0) return false;
        stack<char> s;
        for(auto i:str)
        {
            if(i=='{'||i=='('||i=='[') s.push(i);
            else if(s.empty()) return false;
            else if(i=='}' && s.top()=='{' ) s.pop();
            else if(i==')' && s.top()=='(' ) s.pop();
            else if(i==']' && s.top()=='[' ) s.pop();
            else return false;
        }
        return s.empty();
    }
};

File: Valid Paranthesis.java
import java.util.Stack;
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack=new Stack<>();
        for(char i:s.toCharArray())
        {
            if(i=='{'||i=='['||i=='(')
            stack.push(i);
            else if(stack.isEmpty())
            return false;
            else if(i=='}' && stack.peek()=='{') stack.pop();
            else if(i==']' && stack.peek()=='[') stack.pop();
            else if(i==')' && stack.peek()=='(') stack.pop();
            else return false;
        }
        return stack.isEmpty();
    }
}
